modules/social_features/social_emoji/social_emoji.module

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * @file
+ * The social_emoji module.
+ */
+
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_field_widget_single_element_text_WIDGET_TYPE_form_alter().
+ *
+ * Add emoji trigger element to posts and comments.
+ */
+function social_emoji_field_widget_single_element_text_textarea_form_alter(array &$element, FormStateInterface &$form_state, array $context): void {
+  $element['#suffix'] = "<span class='emoji-trigger'>" . t("Emoji") . "</span><span class='tooltip emoji-picker' role='tooltip'></span>";
```

Comment:
I'm wondering if we can make this in to a FieldWidget setting maybe even extending `TextareaWidget` or something similar.

Why I am wondering is because if right now somebody using Open Social has a textarea or textarea with summary as used in the below form alter hook, the emoji picker is always added. I think we would need to think about the scenario where site builders do not want to have this picker enabled on a given textarea / textarea_with_summary.

Adding a setting to the widget allows people to turn it off / on for their widgets by going to the form display of the textarea / textarea with summary. We can use config to add it by default to our post / comment ones.

<img width="1584" alt="Screenshot 2022-05-04 at 11 36 29" src="https://user-images.githubusercontent.com/16667281/166803135-058bac85-2177-4ba4-bafd-0e4655429fe2.png">
<img width="1583" alt="Screenshot 2022-05-04 at 11 36 39" src="https://user-images.githubusercontent.com/16667281/166803146-b539f956-2fbb-49ee-b09c-2b20476e59a5.png">